### ISSUE TITLE
Support Kinesis UpdateShardCount

### ISSUE THREAD

**Issue by aventurella** [localstack/localstack#654]
https://docs.aws.amazon.com/kinesis/latest/APIReference/API_UpdateShardCount.html

When [Terraform]() creates a kinesis resource it invokes this API to apply the shard count value. It can effectively do nothing in localstack, but it makes working with Terraform nicer as you set up your resources to move it a live AWS instance.

**Terraform output** (only here to show the requests it's making)
```
2018-03-08T14:44:23.489-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: 2018/03/08 14:44:23 [DEBUG] [aws-sdk-go] DEBUG: Request kinesis/UpdateShardCount Details:
2018-03-08T14:44:23.489-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: ---[ REQUEST POST-SIGN ]-----------------------------
2018-03-08T14:44:23.489-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: POST / HTTP/1.1
2018-03-08T14:44:23.489-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: Host: localhost:4568
2018-03-08T14:44:23.489-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: User-Agent: aws-sdk-go/1.12.75 (go1.9.2; darwin; amd64) APN/1.0 HashiCorp/1.0 Terraform/0.11.3
2018-03-08T14:44:23.489-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: Content-Length: 97
2018-03-08T14:44:23.489-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: Authorization: AWS4-HMAC-SHA256 Credential=1234/20180308/us-west-2/kinesis/aws4_request, SignedHeaders=content-length;content-type;host;x-amz-date;x-amz-target, Signature=bbc4fa0c13cf125f10189e922d349500eca85a492968
2018-03-08T14:44:23.489-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: Content-Type: application/x-amz-json-1.1
2018-03-08T14:44:23.489-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: X-Amz-Date: 20180308T224423Z
2018-03-08T14:44:23.489-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: X-Amz-Target: Kinesis_20131202.UpdateShardCount
2018-03-08T14:44:23.489-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: Accept-Encoding: gzip
2018-03-08T14:44:23.489-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4:
2018-03-08T14:44:23.489-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: -----------------------------------------------------
2018-03-08T14:44:23.494-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: 2018/03/08 14:44:23 [DEBUG] [aws-sdk-go] DEBUG: Response kinesis/UpdateShardCount Details:
2018-03-08T14:44:23.495-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: ---[ RESPONSE ]--------------------------------------
2018-03-08T14:44:23.495-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: HTTP/1.1 400 Bad Request
2018-03-08T14:44:23.495-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: Connection: close
2018-03-08T14:44:23.495-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: Content-Length: 38
2018-03-08T14:44:23.495-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: Access-Control-Allow-Headers: authorization,content-type,content-md5,x-amz-content-sha256,x-amz-date,x-amz-security-token,x-amz-user-agent
2018-03-08T14:44:23.495-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: Access-Control-Allow-Methods: HEAD,GET,PUT,POST,DELETE,OPTIONS,PATCH
2018-03-08T14:44:23.495-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: Access-Control-Allow-Origin: *
2018-03-08T14:44:23.495-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: Content-Type: application/x-amz-json-1.1
2018-03-08T14:44:23.495-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: Date: Thu, 08 Mar 2018 22:44:23 GMT
2018-03-08T14:44:23.495-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: Date: Thu, 08 Mar 2018 22:44:23 GMT
2018-03-08T14:44:23.495-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: Server: BaseHTTP/0.3 Python/2.7.10
2018-03-08T14:44:23.495-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: X-Amz-Id-2: Ve5GCkqeQsvR3UP1Hqafo+/1jvit4avpnUwZGAOU20+L4eWZWAbz80iybP+
2018-03-08T14:44:23.495-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: X-Amzn-Requestid: 3f6f7750-2322-11e8-8fbe-cbaf272b93b5
2018-03-08T14:44:23.495-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4:
2018-03-08T14:44:23.495-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4:
2018-03-08T14:44:23.495-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: -----------------------------------------------------
2018-03-08T14:44:23.495-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: 2018/03/08 14:44:23 [DEBUG] [aws-sdk-go] {"__type":"UnknownOperationException"}
2018-03-08T14:44:23.495-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4: 2018/03/08 14:44:23 [DEBUG] [aws-sdk-go] DEBUG: Validate Response kinesis/UpdateShardCount failed, not retrying, error UnknownOperationException:
2018-03-08T14:44:23.495-0800 [DEBUG] plugin.terraform-provider-aws_v1.10.0_x4:  status code: 400, request id: 3f6f7750-2322-11e8-8fbe-cbaf272b93b5
```

**Comment by aventurella** [comment]
ok, so I see that to serve up kinesis functionality Localstack uses 
[kinesalite](https://github.com/mhart/kinesalite).

That does **not** currently support `UpdateShardCount` from what I can tell, I made an issue:

https://github.com/mhart/kinesalite/issues/61
https://github.com/mhart/kinesalite/tree/master/actions

For your consideration I would submit that  since `ProxyListenerKinesis` is a `ProxyListener`, for now, until `kinesalite` supports it, we add something like:

```
ACTION_UPDATE_SHARD_COUNT = '%s.UpdateShardCount' % ACTION_PREFIX
```

in https://github.com/localstack/localstack/blob/master/localstack/services/kinesis/kinesis_listener.py

and the corresponding `elif` check that just returns a mock response like it succeeded.

Per the documentation in `ProxyListener`:

> An instance of requests.models.Response to return to the client instead of the
> actual response returned from the backend service.

**Comment by aventurella** [comment]
```python
elif action == ACTION_UPDATE_SHARD_COUNT:
    response = Response()
    response.status_code = 200

    content = {
        'CurrentShardCount': 1,
        'StreamName': data['StreamName'],
        'TargetShardCount': data['TargetShardCount']
    }
    
    response.encoding = 'UTF-8'
    response.headers = headers
    del response.headers['Content-Length']
    response._content = json.dumps(content)
    return response
```

This would do the trick.

**Comment by nterry** [comment]
@aventurella Are you planning to create the PR? I would love this personally...

**Comment by whummer** [comment]
@aventurella thanks for pointing to the commit in your fork of the repo. Since the primary goal is to make this work with Terraform, I believe it would be acceptable to merge the current "dummy" mock implementation which simply returns a successful response. Would you mind creating a PR for it? Thanks!

**Comment by aventurella** [comment]
Will do.

**Comment by nterry** [comment]
@aventurella I like how you created an [issue with kinesalite](https://github.com/mhart/kinesalite/issues/61). I hope they fix it soon.

**Comment by whummer** [comment]
Should be fixed in #1229 - thanks!

**Comment by semenodm** [comment]
Even though the issue is closed, i think we need to coordinate with the https://github.com/mhart/kinesalite/pull/103 i think instead of stubbing this call we can delegate to the `kinesalite` to do the real shard count update
